File: palindrome.py
import re

regex = re.compile("^[a-z0-9]+$")

def check_palindrome(strinp):

    strinp = strinp.lower()
    l = len(strinp) - 1 
    s = 0
    while s<l:
        first = None
        last = None
        if regex.match(strinp[s]):
            first = strinp[s]
        else:
            s += 1
        
        if regex.match(strinp[l]):
            last = strinp[l]
        else:
            l -= 1

        if first and last:
            if first != last:
                return False

            s += 1
            l -= 1
    return True


if __name__ == "__main__":
    print("\"a used a car\" is a palindrome? %s" % check_palindrome("a used a car"))
    print("\"A man, a plan, a canal: Panama\" is a palindrome? %s" % check_palindrome("A man, a plan, a canal: Panama"))
    print("\"Malayalam\" is a palindrome? %s" % check_palindrome("Malayalam"))
    

File: palindrome.go
package main

import (
	"fmt"
	"regexp"
	"strings"
)

var isLetter = regexp.MustCompile(`^[a-z0-9]+$`).MatchString

func palindrome(str string) bool {
	l := len(str) - 1
	s := 0
	str = strings.ToLower(str)
	for s < l {
		first := byte(0)
		last := byte(0)
		if isLetter(string(str[s])) {
			first = str[s]
		} else {
			s++
		}
		if isLetter(string(str[l])) {
			last = str[l]
		} else {
			l--
		}
		if first != byte(0) && last != byte(0) {

			if first != last {
				return false
			}
			s++
			l--
		}

	}

	return true

}

func main() {
	fmt.Printf("\"a used a car\" is a palindrome? %v\n", palindrome("a used a car"))
	fmt.Printf("\"A man, a plan, a canal: Panama\" is a palindrome? %v\n", palindrome("A man, a plan, a canal: Panama"))
	fmt.Printf("\"Malayalam\" is a palindrome? %v\n", palindrome("Malayalam"))
}
